runtime/doc/treesitter.txt

```diff
@@ -841,6 +841,28 @@ get_files({lang}, {query_name}, {is_included})
         string[] query_files List of files to load for given query and
         language
 
+lint({buf}, {opts})                              *vim.treesitter.query.lint()*
+    Determines diagnostics for a buffer containing treesitter queries
+    (|treesitter-query|). For runtime query files that target installed
+    treesitter parsers, this function will use treesitter API to determine
+    admissible identifiers to describe nodes of those parsers and verifies
+    that top-level s-expressions can be loaded in isolation by Neovim to debug
+    errors in runtime query files. The found diagnostics are reported using
+    |diagnostic-api|.
+
+    Parameters: ~
+      • {buf}   integer Buffer to lint
+      • {opts}  nil|table Options for linting
+                • langs nil|string|string[] One or more parsers which the
+                  current queries target (see |treesitter-parsers|) or a
+                  function that determines the list of parsers given `buf`,
+                  `opts`. By default the query parser is determined using the
+                  containing folder of the query buffer
+                  (`<lang>/<query_type>.scm`, e.g. `cpp` for
+                  `cpp/highlights.scm`)
+                • clear bool when true just clear existing lints instead of
```

Comment:
```suggestion
                • clear (bool) clear existing lints instead of
```

Reply:
(sorry, I always make the mistake of reviewing top to bottom; these fixes should of course go to the docstring)

Reply:
And I want to take a bit of time to edit the description here; I find it _really_ hard to understand...

Reply:
we could also omit this option, assuming it is not needed for most use cases. This option has a bit the style of inverting the original action of the function (`create_file(path, { delete = true })`)